docs/PATCH-ENVELOPES.md

```diff
@@ -1,36 +1,99 @@
+`169.254.169.254`. It has bunch of useful endpoints, amongst them are patch envelope endpoints. So within
+App Instance one can access Patch Envelopes available to specific App Instance by getting description.json.
+This would return list of Patch Envelopes available to this App Instance.
 
-*Inline artifacts* are small (less or equal than 100KB) base64-encoded (not encrypted) strings with
-optional meta data. They are part of Edge Device configuration.
-*External artifacts* are referrencining volumes created on EVE. Size of volumes is not limited to 10KB
+```bash
+curl -X GET -v http://169.254.169.254/eve/v1/patch/description.json
+[
 
-## How to use Patch Envelopes
+    {
+        "PatchId":"699fbdb2-e455-448f-84f5-68e547ec1305",
+        "Version":"1",
+        "BinaryBlobs":[
+            {
+                "file-name":"textfile1.txt",
+                "file-sha":"%FILE_SHA",
+                "file-meta-data":"YXJ0aWZhY3QgbWV0YWRhdGE=",
+                "url":"http://169.254.169.254/eve/v1/patch/download/699fbdb2-e455-448f-84f5-68e547ec1305/textfile1.txt"
+            },
+            {
+                "file-name":"textfile2.txt",
+                "file-sha":"%FILE_SHA%",
+                "file-meta-data":"YXJ0aWZhY3QgbWV0YWRhdGE=",
+                "url":"http://169.254.169.254/eve/v1/patch/download/699fbdb2-e455-448f-84f5-68e547ec1305/textfile2.txt"
+            }
+        ],
+        "VolumeRefs":null
+    }
 
-When Patch Envelopes are created on controller and exposed to EVE via API, app instance can access
-Patch Envelopes available to it from meta-data server using API defined in [metadata server](ECO-METADATA.md)
+] 
+```
+
+Every Patch Envelope contains of one or more Binary Artefacts (Blobs) which are base64-encoded objects.
+Each object can be downloaded by calling URL, i.e. 
+
+```bash
+curl -X GET http://169.254.169.254/eve/v1/patch/download/699fbdb2-e455-448f-84f5-68e547ec1305/textfile1.txt
+```
+
+will get you base64-encoded file.Note that you can download zip archive of all binary artifacts for a
+given patch envelope by calling
+
+```bash
+curl -X GET http://169.254.169.254/eve/v1/patch/download/699fbdb2-e455-448f-84f5-68e547ec1305 > a.zip
+```
+
+Flow diagram of the process is below
+
+![process-flow](./images/eve-pe-process-flow.png)
+
+Full OpenAPI (Swagger) specification for patch envelope endpoint can be found [here](./api/patch-envelopes.yml).
+You can generate client from this specification and use it to develop your application.
+
+## What types of Binary Artifacts are there?
+
+There’re two types of Binary Artifacts (Blobs): inline and external. When user
+download Binary Artifact there’s no difference when it’s downloaded. However,
+there’s a difference on how this artifacts are treated in EVE. Inline binary
+artifacts are small size artifacts (10KBytes) which are part of EdgeDevConfig,
+whereas external patch envelopes are represented as Volumes which are handled
+by volumemgr. 
```

Comment:
You can maybe link pkg/pillar/docs/volumemgr.md

Reply:
Done